test/osb_test/osb_test.go

```diff
@@ -0,0 +1,322 @@
+	}).Methods("PUT")
+
+	router.HandleFunc("/good/v2/service_instances/{instance_id}/service_bindings/{binding_id}", func(rw http.ResponseWriter, req *http.Request) {
+		setResponse(rw, http.StatusOK, "{}")
+	}).Methods("DELETE")
+
+	router.PathPrefix("/bad/").HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		setResponse(rw, http.StatusNotAcceptable, `{"description": "expected error"}`)
+	})
+
+	return router
+}
+
+func registerBroker(brokerJSON object, SM *httpexpect.Expect) string {
+	reply := SM.POST("/v1/service_brokers").
+		WithJSON(brokerJSON).
+		Expect().Status(http.StatusCreated).JSON().Object()
+	return reply.Value("id").String().Raw()
+}
+
+func assertBadBrokerError(req *httpexpect.Request) {
+	body := req.Expect().Status(http.StatusNotAcceptable).JSON().Object()
+	body.ContainsKey("description")
+	body.Value("description").Equal("expected error")
+}
+
+func assertMissingBrokerError(req *httpexpect.Request) {
+	body := req.Expect().Status(http.StatusNotFound).JSON().Object()
+	body.ContainsKey("description").
+		Value("description").String().
+		Equal("Could not find broker with id: missing_broker_id")
+}
+
+func getDummyService(idsToRemove ...string) *object {
+	result := &object{
+		"service_id":        "dummyId",
+		"plan_id":           "dummyplanId",
+		"organization_guid": "orgguid",
+		"space_guid":        "spaceguid",
+	}
+	for _, id := range idsToRemove {
+		delete(*result, id)
+	}
+	return result
+}
+
+type smreq func(path string, pathargs ...interface{}) *httpexpect.Request
+
+func assertRequiredIDError(resp *httpexpect.Object, expectedIDName string) {
+	resp.ContainsKey("description").
+		Value("description").String().
+		Equal(expectedIDName + " is required")
+}
+
+func requestWithMissingIDsInBody(req smreq, url string, idName string, expectedIDName string) {
+	resp := req(url).WithHeader("X-Broker-API-Version", "2.13").
+		WithJSON(getDummyService(idName)).
+		Expect().Status(http.StatusBadRequest).JSON().Object()
+
+	assertRequiredIDError(resp, expectedIDName)
+}
+
+func requestWithMissingIDsInQuery(req smreq, url string, idName string, expectedIDName string) {
+	resp := req(url).WithHeader("X-Broker-API-Version", "2.13").
+		WithQueryObject(getDummyService(idName)).
+		Expect().Status(http.StatusBadRequest).JSON().Object()
+
+	assertRequiredIDError(resp, expectedIDName)
+}
+
+// TestOSB tests for OSB API
+func TestOSB(t *testing.T) {
+	os.Chdir("../..")
+	RunSpecs(t, "OSB API Tests Suite")
+}
+
+var _ = Describe("Service Manager OSB API", func() {
+	var SM *httpexpect.Expect
+	var testServer *httptest.Server
+
+	var listener net.Listener
+	var goodBrokerID string
+	var badBrokerID string
+
+	BeforeSuite(func() {
+		testServer = httptest.NewServer(common.GetServerRouter())
+		SM = httpexpect.New(GinkgoT(), testServer.URL)
+		common.RemoveAllBrokers(SM)
+		listener, _ = net.Listen("tcp", ":0")
+		brokerRouter := createBrokerRouter()
+		go http.Serve(listener, brokerRouter)
```

Comment:
would it make sense to use httptest.NewServer here as well to start a different server for the broker instead of go http.Serve?